tools/autograd/templates/Functions.cpp

```diff
@@ -809,10 +799,545 @@ Tensor softplus_double_backward(const Tensor & grad, const Tensor & input, Scala
   return _sigmoid_backward(grad, x.sigmoid()) * (x < threshold).toType(grad.type()) * beta;
 }
 
-Tensor as_strided_backward(const Tensor & grad, TensorGeometry base, IntList sizes, IntList strides, int64_t storage_offset) {
-  auto src = base.zeros_with_stride(grad.type());
-  src.as_strided(sizes, strides, storage_offset - base.storage_offset()).copy_(grad);
-  return src;
+
+// NOTE [ as_strided Backward ]
```

Comment:
There are two notes:
1. This one explaining how `as_strided_backward` works. It explains why we want to know whether input and output geometry has overlapping memory, and how do we use these information to compute a backward that is compatible with the rest of our autograd system (basically we need to make `as_strided` "look" stride-agnostic).
2. There is one more below on how to detect memory overlap. It checks for a constrained simple case, but should be sufficient on all combination of our view ops except for `unfold` and general `as_stride` (proofs included).